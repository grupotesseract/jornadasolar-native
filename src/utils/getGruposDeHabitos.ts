import { t } from 'i18n-js'
import GrupoDeHabitos from '../entities/GrupoDeHabitos'
import GetAllGruposDeHabitosModelos from '../services/gruposDeHabitos/GetAllGruposDeHabitosModelos'
import GetUserGruposDeHabitos from '../services/user/GetUserGruposDeHabitos'

const getGruposDeHabitosIniciais = async (
  userId?: string
): Promise<GrupoDeHabitos[]> => {
  const grupoDeHabitosModelos = userId
    ? await GetUserGruposDeHabitos(userId)
    : tiraGrupoPersonalizado(await new GetAllGruposDeHabitosModelos().call())
  return grupoDeHabitosModelos
}

const getGruposDeHabitosTemplate = async (
  userId?: string
): Promise<GrupoDeHabitos[]> => {
  let grupo = await getGruposDeHabitosIniciais(userId)
  grupo = grupo.map(grupo => {
    return { id: grupo.id, nome: grupo.nome, habitos: [] }
  })
  return grupo
}

const tiraGrupoPersonalizado = (grupos: GrupoDeHabitos[]) => {
  return grupos.filter(
    grupo =>
      grupo.nome !== t('comum.nomeGrupoPersonalizado') &&
      grupo.nome !== 'Personalizados'
  )
}

export { getGruposDeHabitosTemplate, getGruposDeHabitosIniciais }
