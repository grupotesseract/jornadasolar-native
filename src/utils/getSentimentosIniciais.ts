import Sentimento from '../entities/Sentimento'
import GetAllSentimentosModelos from '../services/sentimentosModelos/GetAllSentimentosModelos'
import GetUserSentimentos from '../services/user/GetUserSentimentos'

const getSentimentosIniciais = async (
  userId?: string
): Promise<Sentimento[]> => {
  const sentimentos = userId
    ? await new GetUserSentimentos(userId).call()
    : await new GetAllSentimentosModelos().call()
  return sentimentos
}

export default getSentimentosIniciais
