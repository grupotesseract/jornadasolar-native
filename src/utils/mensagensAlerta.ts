import i18n from '../i18n'

enum TiposAlerta {
  Sucesso = 'Sucesso',
  Falha = 'Falha'
}

interface AlertProps {
  tipo: TiposAlerta
  texto: string
}

const { t } = i18n

const registroSucesso: AlertProps = {
  tipo: TiposAlerta.Sucesso,
  texto: t('mensagensAlerta.registroSucesso')
}

const registroFalha: AlertProps = {
  tipo: TiposAlerta.Falha,
  texto: t('mensagensAlerta.registroFalha')
}

const atualizacaoSucesso = (item: string): AlertProps => {
  return {
    tipo: TiposAlerta.Sucesso,
    texto: t('mensagensAlerta.atualizacaoSucesso', { tipo: item.toLowerCase() })
  }
}
const atualizacaoFalha = (item: string): AlertProps => {
  return {
    tipo: TiposAlerta.Falha,
    texto: t('mensagensAlerta.atualizacaoFalha', { tipo: item.toLowerCase() })
  }
}
const criacaoFalha = (item: string): AlertProps => {
  return {
    tipo: TiposAlerta.Falha,
    texto: t('mensagensAlerta.criacaoFalha', { tipo: item.toLowerCase() })
  }
}
const senhaAlteradaSucesso = (): AlertProps => {
  return {
    tipo: TiposAlerta.Sucesso,
    texto: t('mensagensAlerta.alteracaoSenhaSucesso')
  }
}

export {
  TiposAlerta,
  AlertProps,
  registroSucesso,
  registroFalha,
  atualizacaoSucesso,
  atualizacaoFalha,
  criacaoFalha,
  senhaAlteradaSucesso
}
