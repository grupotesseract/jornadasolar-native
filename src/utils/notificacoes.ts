import * as Notifications from 'expo-notifications'
import Constants from 'expo-constants'
import { Platform } from 'react-native'
import add from 'date-fns/add'
import { t } from 'i18n-js'
import { format } from 'date-fns'

Notifications.setNotificationHandler({
  handleNotification: async () => ({
    shouldShowAlert: true,
    shouldPlaySound: false,
    shouldSetBadge: false
  })
})

async function agendaNotificacaoTresDias() {
  const date = add(new Date(), { days: 3 })
  await Notifications.scheduleNotificationAsync({
    content: {
      title: t('nomeApp'),
      body: t('notificacoes.tresDias'),
      data: {
        link: 'dia',
        params: {
          data: format(date, 'd-M-yyyy')
        }
      }
    },
    trigger: {
      date
    }
  })
}

async function registraTokenParaNotificacoesExternas() {
  let token: string
  if (Constants.isDevice) {
    const { status: existingStatus } = await Notifications.getPermissionsAsync()
    let finalStatus = existingStatus
    if (existingStatus !== 'granted') {
      const { status } = await Notifications.requestPermissionsAsync()
      finalStatus = status
    }
    if (finalStatus !== 'granted') {
      console.log('Erro ao registrar token para notificações push')
      return
    }
    token = (await Notifications.getExpoPushTokenAsync())?.data
  }

  if (Platform.OS === 'android') {
    Notifications.setNotificationChannelAsync('Jornada', {
      name: 'Jornada Solar',
      importance: Notifications.AndroidImportance.MAX
    })
  }
  return token
}

async function cancelaNotificacoesAgendadas() {
  await Notifications.cancelAllScheduledNotificationsAsync()
}

export {
  agendaNotificacaoTresDias,
  registraTokenParaNotificacoesExternas,
  cancelaNotificacoesAgendadas,
  Notifications
}
