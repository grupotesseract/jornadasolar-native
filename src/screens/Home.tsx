import React from 'react'
import { HomeNavigationProps } from '../routes/Home.routes'
import { View, StyleSheet, Image } from 'react-native'
import { Caption, Text, Title } from 'react-native-paper'
import { appVersion } from '../utils/appVersion'
import Button from '../components/Button'
import Icone from '../../assets/icon.png'
import { theme } from '../../theme'
import i18n from '../i18n'

export default function Home({ navigation }: HomeNavigationProps) {
  const { t } = i18n

  const handleCadastro = () => {
    navigation.navigate('Identificacao')
  }

  const handleLogin = () => {
    navigation.navigate('Login')
  }

  return (
    <View style={styles.container}>
      <View style={styles.centralizado}>
        <Title style={styles.cabecalho}>{t('nomeApp')}</Title>
        <Caption style={styles.cabecalho}>
          {t('home.versao')} {appVersion}
        </Caption>
      </View>
      <View style={styles.centralizado}>
        <Image source={Icone} style={styles.imagem} />
        <Text style={styles.frase}>{t('home.frase')}</Text>
      </View>
      <View style={styles.botoes}>
        <View style={styles.botao}>
          <Button onPress={handleCadastro} testID="botaoCadastro">
            {t('home.comecarJornada')}
          </Button>
        </View>
        <View style={styles.botao}>
          <Button mode="outlined" onPress={handleLogin} testID="botaoLogin">
            {t('home.tenhoCadastro')}
          </Button>
        </View>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 20,
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  centralizado: {
    alignItems: 'center'
  },
  cabecalho: {
    color: theme.colors.primary
  },
  imagem: {
    width: 134,
    height: 134
  },
  frase: {
    fontWeight: '400',
    fontSize: 20,
    lineHeight: 28,
    textAlign: 'center',
    marginTop: 30
  },
  botao: {
    marginBottom: 16
  },
  botoes: {
    width: '90%'
  }
})
