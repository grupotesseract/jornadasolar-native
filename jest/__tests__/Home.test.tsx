import React from 'react'
import Home from '../../src/screens/Home'
import { fireEvent, render } from '@testing-library/react-native'
import { navigationTestProps } from './utils'

describe('Teste de componente: Tela Home', () => {
  let props: any
  let home: any
  beforeEach(() => {
    props = navigationTestProps({})
    home = render(<Home {...props} />)
  })

  test('renderiza igual ao último snapshot', () => {
    expect(home.toJSON()).toMatchSnapshot()
  })

  test('botão Login chama navegação para Login', () => {
    const { getByTestId } = home

    const botaoLogin = getByTestId('botaoLogin')
    fireEvent.press(botaoLogin)

    expect(botaoLogin).toBeTruthy()
    expect(props.navigation.navigate).toHaveBeenCalledWith('Login')
  })
  test('botão Cadastro chama navegação para Identificação', () => {
    const { getByTestId } = home

    const botaoCadastro = getByTestId('botaoCadastro')
    expect(botaoCadastro).toBeTruthy()

    fireEvent.press(botaoCadastro)
    expect(props.navigation.navigate).toHaveBeenCalledWith('Identificacao')
  })
})
