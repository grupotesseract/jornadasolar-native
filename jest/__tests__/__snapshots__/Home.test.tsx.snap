// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de componente: Tela Home renderiza igual ao último snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              Object {
                "color": "#F7C92A",
              },
            ],
          ],
        ]
      }
    >
      Jornada Solar
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 12,
                "letterSpacing": 0.4,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              Object {
                "color": "#F7C92A",
              },
            ],
          ],
        ]
      }
    >
      versão
       
      1.1.0
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 134,
          "width": 134,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "fontSize": 20,
            "fontWeight": "400",
            "lineHeight": 28,
            "marginTop": 30,
            "textAlign": "center",
          },
        ]
      }
    >
      Embarque em uma jornada de autoconhecimento e monitore seus hábitos e emoções
    </Text>
  </View>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#6200ee",
            "borderColor": "transparent",
            "borderRadius": 100,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "justifyContent": "center",
            "minWidth": 64,
            "padding": 6,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityLabel="botaoCadastro"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 100,
              },
            ]
          }
          testID="botaoCadastro"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontSize": 20,
                      "fontWeight": "400",
                      "textTransform": "none",
                    },
                  ],
                ]
              }
            >
              Começar Jornada
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#F7C92A",
            "borderRadius": 100,
            "borderStyle": "solid",
            "borderWidth": 2,
            "elevation": 0,
            "justifyContent": "center",
            "minWidth": 64,
            "padding": 6,
          }
        }
      >
        <View
          accessibilityLabel="botaoLogin"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 100,
              },
            ]
          }
          testID="botaoLogin"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontSize": 20,
                      "fontWeight": "400",
                      "textTransform": "none",
                    },
                  ],
                ]
              }
            >
              Já tenho cadastro
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
